src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala

```diff
@@ -85,7 +87,57 @@ final case class PurposesApiServiceImpl(
     onComplete(result) {
       handleError(
         s"Error retrieving Purposes for name $q, EServices $eServicesIds, Consumers $consumersIds offset $offset, limit $limit"
-      ) orElse { case Success(r) => getPurposes200(r) }
+      ) orElse { case Success(r) => getConsumerPurposes200(r) }
+    }
+  }
+
+  override def getProducerPurposes(
```

Comment:
the two endpoints are identical, let's write a single parametric function called by both and remove code duplication